Std/Data/Bool.lean

```diff
@@ -0,0 +1,251 @@
+
+theorem xor_not_not : ∀ (x y : Bool), xor (!x) (!y) = (xor x y) := by
+  decide
+
+theorem xor_self : ∀ (x : Bool), xor x x = false := by
+  decide
+
+theorem xor_comm : ∀ (x y : Bool), xor x y = xor y x := by
+  decide
+
+theorem xor_left_comm : ∀ (x y z : Bool), xor x (xor y z) = xor y (xor x z) := by
+  decide
+
+theorem xor_right_comm : ∀ (x y z : Bool), xor (xor x y) z = xor (xor x z) y := by
+  decide
+
+theorem xor_assoc : ∀ (x y z : Bool), xor (xor x y) z = xor x (xor y z) := by
+  decide
+
+/-! ### le/lt -/
+
+@[simp] protected theorem le_true : ∀ (x : Bool), x ≤ true := by
+  decide
+
+@[simp] protected theorem false_le : ∀ (x : Bool), false ≤ x := by
+  decide
+
+@[simp] protected theorem le_refl : ∀ (x : Bool), x ≤ x := by
+  decide
+
+@[simp] protected theorem lt_irrefl : ∀ (x : Bool), ¬ x < x := by
+  decide
+
+protected theorem le_trans : ∀ {x y z : Bool}, x ≤ y → y ≤ z → x ≤ z := by
+  decide
+
+protected theorem le_antisymm : ∀ {x y : Bool}, x ≤ y → y ≤ x → x = y := by
+  decide
+
+protected theorem le_total : ∀ (x y : Bool), x ≤ y ∨ y ≤ x := by
+  decide
+
+protected theorem lt_asymm : ∀ {x y : Bool}, x < y → ¬ y < x := by
+  decide
+
+protected theorem lt_trans : ∀ {x y z : Bool}, x < y → y < z → x < z := by
+  decide
+
+protected theorem lt_iff_le_not_le : ∀ {x y : Bool}, x < y ↔ x ≤ y ∧ ¬ y ≤ x := by
+  decide
+
+protected theorem lt_of_le_of_lt : ∀ {x y z : Bool}, x ≤ y → y < z → x < z := by
+  decide
+
+protected theorem lt_of_lt_of_le : ∀ {x y z : Bool}, x < y → y ≤ z → x < z := by
+  decide
+
+protected theorem le_of_lt : ∀ {x y : Bool}, x < y → x ≤ y := by
+  decide
+
+protected theorem le_of_eq : ∀ {x y : Bool}, x = y → x ≤ y := by
+  decide
+
+protected theorem ne_of_lt : ∀ {x y : Bool}, x < y → x ≠ y := by
+  decide
+
+protected theorem lt_of_le_of_ne : ∀ {x y : Bool}, x ≤ y → x ≠ y → x < y := by
+  decide
+
+protected theorem le_of_lt_or_eq : ∀ {x y : Bool}, x < y ∨ x = y → x ≤ y := by
+  decide
+
+protected theorem eq_true_of_true_le : ∀ {x : Bool}, true ≤ x → x = true := by
+  decide
+
+protected theorem eq_false_of_le_false : ∀ {x : Bool}, x ≤ false → x = false := by
+  decide
+
+/-! ### min/max -/
+
+protected theorem max_eq_or : ∀ (x y : Bool), max x y = (x || y) := by
+  decide
+
+protected theorem min_eq_and : ∀ (x y : Bool), min x y = (x && y) := by
+  decide
+
+/-! ### injectivity lemmas -/
+
+theorem not_inj : ∀ {x y : Bool}, (!x) = (!y) → x = y := by
+  decide
+
+theorem not_inj' : ∀ {x y : Bool}, (!x) = (!y) ↔ x = y := by
+  decide
+
+theorem and_or_inj_right : ∀ {m x y : Bool}, (x && m) = (y && m) → (x || m) = (y || m) → x = y := by
+  decide
+
+theorem and_or_inj_right' :
+    ∀ {m x y : Bool}, (x && m) = (y && m) ∧ (x || m) = (y || m) ↔ x = y := by
+  decide
+
+theorem and_or_inj_left : ∀ {m x y : Bool}, (m && x) = (m && y) → (m || x) = (m || y) → x = y := by
+  decide
+
+theorem and_or_inj_left' : ∀ {m x y : Bool}, (m && x) = (m && y) ∧ (m || x) = (m || y) ↔ x = y := by
+  decide
+
+end Bool
```

Comment:
I think for consistency, let's remove the `end Bool`.

This could eventually be a lint check or something an auto-formatter resolves.